demo/js/application.js

```diff
@@ -4,52 +4,165 @@ let readline = require('readline-sync');
+                this._printer.paragrapheStartMessage('Please pick the book\'s id or type return');
+
+                id = readline.question('');
+                if (id === 'return')
+                    return true;
+
+                bookfound = this._library.getBookById(id);
+
+                if (bookfound === null){
+                    this._printer.warningMessage('No book with this id exists! Try again.');
+                }
+            }
+
+            this._printer.paragrapheStartMessage('Please choose a letter to pick an option : ');
+            this._printer.basicMessage('I - Update a book\'s isbn.');
+            this._printer.basicMessage('T - Update a book\'s title.');
+            this._printer.basicMessage('D - Update a book\'s description.');
+            this._printer.basicMessage('return - Go back to main menu.');
+
+            response = readline.question('');
+
+            switch(response){
+            case 'I':
+                this._printer.paragrapheStartMessage('Please type the book\'s new isbn : ');
+                isbn = readline.question('');
+
+                if (!this._library.updateBookIsbn(id, isbn)){
+                    this._printer.paragrapheStartMessage('This isbn already exists.');
+                }
+                return true;
+            case 'T':
+                this._printer.paragrapheStartMessage('Please type the book\'s new title : ');
+                title = readline.question('');
+
+                if (!this._library.updateBookTitle(id, title)){
+                    this._printer.paragrapheStartMessage('This title already exists.');
+                }
+                return true;
+            case 'D':
+                this._printer.paragrapheStartMessage('Please type the book\'s new description : ');
+                description = readline.question('');
+
+                this._library.updateBookDescription(id, description);
+                return true;
+            case 'return':
+                return true;
+            default :
+                this._printer.warningMessage('Your choice is not an option');
+                break;
+            }
+        }
+    }
+
+    deleteBookChoice(){
+        let id = '';
+        let bookfound = null;
+
+        while (bookfound === null){
+            this._printer.paragrapheStartMessage('Please pick the book\'s id or type return');
+
+            id = readline.question('');
+            if (id === 'return')
+                return true;
+
+            bookfound = this._library.getBookById(id);
+
+            if (bookfound === null){
+                this._printer.warningMessage('No book with this id exists! Try again.');
+            }
+        }
+
+        if (this._library.deleteBookById(id)){
+            this._printer.paragrapheStartMessage('The book has been deleted.');
+        }
+        else {
+            this._printer.warningMessage('An error occurred while trying to delete.');
+            return false;
```

Comment:
use an early return. Just move the content of the else in a reverse if and get the content of the if outside the conditional.